### ISSUE TITLE
Redis authentication password

### ISSUE THREAD

**Issue by michelepatrassi** [nestjs/nest#4915]
## Is your feature request related to a problem? Please describe.
I want to connect to "Azure Cache for Redis", which provides host, port and access key. Which the standard nodejs `redis` library, I can connect like this 
```
var redis = require ('redis');
  var client = redis.createClient ('PORT', 'HOST', {
    auth_pass: 'ACCESS_KEY',
    tls: {servername: 'HOST'},
  });
```
While in NestJS I can only configure the url https://docs.nestjs.com/microservices/redis#overview. I did not found anything regarding passing the password/key in the URL itself. I think this is a general issue with every non-local redis instance, so that's why I'm asking the feature (or current solution if I'm wrong).

## Describe the solution you'd like
I would like to instantiate the connection like this
```
const app = await NestFactory.createMicroservice<MicroserviceOptions>(ApplicationModule, {
  transport: Transport.REDIS,
  options: {
    url: 'redis://localhost:6379',
    authPass: '1234'
  },
});
```

## What is the motivation / use case for changing the behavior?
Connect with an external redis instance by providing a password/credentials

**Comment by kamilmysliwiec** [comment]
We should rather allow passing any `ClientOpts` (e.g., `socket_nodelay`, `parser`, `tls`) properties as "options" instead of adding them manually to the class.

**Comment by michelepatrassi** [comment]
@kamilmysliwiec you are completely right, that's also why I moved the PR to draft. I privately patched the compiled js like this for now
```
     createRedisClient() {
+        return redisPackage.createClient(Object.assign(Object.assign({}, this.getClientOptions()), {   
+            url: this.url,
+            auth_pass: this.getOptionsProp(this.options, 'authPass'),
+            tls: this.getOptionsProp(this.options, 'tls') 
+        }));
     }
```
which is ok if you want to preserve the camelcase parameters, but it may be better as long term solution to simply pass an object which supports all the official client options https://www.npmjs.com/package/redis#options-object-properties and refer it in the official docs.

**Comment by kamilmysliwiec** [comment]
> but it may be better as long term solution to simply pass an object which supports all the official client options https://www.npmjs.com/package/redis#options-object-properties and refer it in the official docs.

Fixed in 7.2.0. Let's track the docs PR here https://github.com/nestjs/docs.nestjs.com/issues/1300
